src/prefect/results.py

```diff
@@ -103,13 +103,17 @@ async def get_storage() -> WritableFileSystem:
             )
 
         try:
-            await block.save(name, overwrite=False)
-            return block
+            block = await block.save(name, overwrite=False)
         except ValueError as e:
             if "already in use" not in str(e):
                 raise e
+        except ObjectAlreadyExists:
+            # Another client created the block before we reached this line
+            pass
+        else:
+            block = await Block.load(default_storage_name)
```

Comment:
Doesn't this `else` only run if the `ObjectAlreadyExists` exception is _not_ raised?  Meaning you'd want this load to happen within the new except?

Reply:
+1, I think if we did hit `ObjectAlreadyExists` here, the `block` we return would be the unsaved `LocalFileSystem` block we created on L97, right?

Reply:
Ah yeah, I was tired 😂

Reply:
Nevertheless I continue to believe except/else wasn't a mistaken language feature